examples/kepler-integration/src/reducers/index.js

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {combineReducers} from 'redux';
+import {handleActions} from 'redux-actions';
+
+import keplerGlReducer, {combinedUpdaters, uiStateUpdaters} from 'kepler.gl/reducers';
+import {processGeojson, processCsvData} from 'kepler.gl/processors';
+import KeplerGlSchema from 'kepler.gl/schemas';
+import {EXPORT_MAP_FORMATS} from 'kepler.gl/constants';
+
+import {
+  INIT,
+  LOAD_MAP_SAMPLE_FILE,
+  LOAD_REMOTE_RESOURCE_SUCCESS,
+  LOAD_REMOTE_RESOURCE_ERROR,
+  SET_SAMPLE_LOADING_STATUS
+} from '../actions';
+
+import {AUTH_TOKENS, DEFAULT_FEATURE_FLAGS} from '../constants/default-settings';
+import {generateHashId} from '../utils/strings';
+
+// INITIAL_APP_STATE
+const initialAppState = {
+  appName: 'example',
+  loaded: false,
+  sampleMaps: [], // this is used to store sample maps fetch from a remote json file
+  isMapLoading: false, // determine whether we are loading a sample map,
+  error: null, // contains error when loading/retrieving data/configuration
+  // {
+  //   status: null,
+  //   message: null
+  // }
+  // eventually we may have an async process to fetch these from a remote location
+  featureFlags: DEFAULT_FEATURE_FLAGS
+};
+
+// App reducer
+export const appReducer = handleActions(
+  {
+    [INIT]: state => ({
+      ...state,
+      loaded: true
+    }),
+    [LOAD_MAP_SAMPLE_FILE]: (state, action) => ({
+      ...state,
+      sampleMaps: action.samples
+    }),
+    [SET_SAMPLE_LOADING_STATUS]: (state, action) => ({
+      ...state,
+      isMapLoading: action.isMapLoading
+    })
+  },
+  initialAppState
+);
+
+const {DEFAULT_EXPORT_MAP} = uiStateUpdaters;
+
+// combine app reducer and keplerGl reducer
+// to mimic the reducer state of kepler.gl website
+const demoReducer = combineReducers({
+  // mount keplerGl reducer
+  keplerGl: keplerGlReducer.initialState({
```

Comment:
This is the app you're testing and connecting within, right? It looks like the right direction. I wonder if there's a small thing causing this object to look empty.

Reply:
This is part of the template demo app. I haven't gotten to this yet. In the future I'll likely be putting my custom hubble modal action in there.